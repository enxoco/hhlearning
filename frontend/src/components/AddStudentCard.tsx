import { Alert, AlertIcon, AlertTitle, Box, Button, Divider, Flex, FormControl, FormLabel, Input, Stack, useColorModeValue, useToast } from "@chakra-ui/react"
import { useState } from "react"
import { useCreateStudentMutation, useUpdateStudentInfoMutation } from "../generated/graphql"
import {useParams} from 'react-router-dom'

const AddStudentCard = ({student}) => {

  const [firstName, setFirstName] = useState(student?.firstName || null)
  const [lastName, setLastName] = useState(student?.lastName || null)
  const [{ data, error }, addStudent] = useCreateStudentMutation()
  const [updatedStudentInfo, updateStudentInfo] = useUpdateStudentInfoMutation()
  let { id } = useParams()
  const toast = useToast({ position: 'top', isClosable: true, duration: 9000 });
  const handleFirstNameUpdate = (e) => {
    setFirstName(e.target.value)
  }

  const handleLastNameUpdate = (e) => {
    setLastName(e.target.value)
  }

  const handleFormSubmit = (e) => {
      e.preventDefault()
    if (!student) {
        addStudent({ firstName, lastName })
        if (error) {
          console.error("error", error)
          return
        }

    } else {
        updateStudentInfo({
          id: id,
          firstName: firstName,
          lastName: lastName,
          isFormer: false
        })
    }
    toast({
      title: 'Success',
      status: 'success',
      description: 'Student updated successfully'
    })

  }

  return (
    <Box bg="bg-surface" boxShadow={useColorModeValue("sm", "sm-dark")} borderRadius="lg">
    {data?.createStudent ? (
      <Alert status="success">
        <AlertIcon />
        <AlertTitle>Student added successfully</AlertTitle>
      </Alert>
    ) : null}
    <Stack spacing="5" px={{ base: "4", md: "6" }} py={{ base: "5", md: "6" }}>
      <Stack spacing="6" direction={{ base: "column", md: "row" }}>
        <FormControl id="firstName">
          <FormLabel>First Name</FormLabel>
          <Input defaultValue={firstName} onChange={handleFirstNameUpdate} />
        </FormControl>
        <FormControl id="lastName" placeholder="">
          <FormLabel>Last Name</FormLabel>
          <Input value={lastName} onChange={handleLastNameUpdate} />
        </FormControl>
      </Stack>
    </Stack>
    <Divider />
    <Flex direction="row-reverse" py="4" px={{ base: "4", md: "6" }}>
      <Button variant="primary" onClick={handleFormSubmit}>
        Save
      </Button>
    </Flex>
  </Box>
  )
}

export default AddStudentCard
