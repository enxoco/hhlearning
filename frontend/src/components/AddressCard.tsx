import {
    Box,
    BoxProps,
    Button,
    Divider,
    Flex,
    FormControl,
    FormLabel,
    Input,
    Stack,
    useColorModeValue,
  } from '@chakra-ui/react'
  import {useCheckLoginQuery, useGetMyProfileQuery} from "#/generated/graphql"

  export const PersonalInfoCard = (props: BoxProps) => {
    const [me] = useCheckLoginQuery()
    const [profile] = useGetMyProfileQuery({variables: {id: me.data?.authenticatedItem.id}, pause: !me.data?.authenticatedItem})
    const {name, firstName, lastName, city, street, state, zipcode} = profile.data?.user
      return (
        <Box
          as="form"
          bg="bg-surface"
          boxShadow={useColorModeValue('sm', 'sm-dark')}
          borderRadius="lg"
          {...props}
        >
          <Stack spacing="5" px={{ base: '4', md: '6' }} py={{ base: '5', md: '6' }}>
            <Stack spacing="6" direction={{ base: 'column', md: 'row' }}>
              <FormControl id="firstName">
                <FormLabel>First Name</FormLabel>
                <Input defaultValue={firstName || name} />
              </FormControl>
              <FormControl id="lastName">
                <FormLabel>Last Name</FormLabel>
                <Input defaultValue={lastName} />
              </FormControl>
            </Stack>
            <FormControl id="street">
              <FormLabel>Street</FormLabel>
              <Input defaultValue={street} />
            </FormControl>
            <Stack spacing="6" direction={{ base: 'column', md: 'row' }}>
              <FormControl id="city">
                <FormLabel>City</FormLabel>
                <Input defaultValue={city} />
              </FormControl>
              <FormControl id="state">
                <FormLabel>State / Province</FormLabel>
                <Input defaultValue={state} />
              </FormControl>
              <FormControl id="zip">
                <FormLabel>ZIP/ Postal Code</FormLabel>
                <Input defaultValue={zipcode} />
              </FormControl>
            </Stack>
          </Stack>
          <Divider />
          <Flex direction="row-reverse" py="4" px={{ base: '4', md: '6' }}>
            <Button type="submit" variant="primary">
              Save
            </Button>
          </Flex>
        </Box>
      )
  }