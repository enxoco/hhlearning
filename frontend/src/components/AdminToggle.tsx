import { User, useToggleAdminMutation } from "#/generated/graphql";
import { Button, FormControl, HStack } from "@chakra-ui/react";

export default function(user: User){
  const [toggleStatus, toggleAdmin] = useToggleAdminMutation();

    return (
        <HStack spacing="1">
          <FormControl display="flex" alignItems="center">
              <Button disabled={toggleStatus.fetching} colorScheme={user.isAdmin ? "blue" : "gray"} borderRadius={0} onClick={() => toggleAdmin({ id: user.id, isAdmin: true})}>Yes</Button>
              <Button disabled={toggleStatus.fetching} colorScheme={!user.isAdmin ? "blue" : "gray"} borderRadius={0} onClick={() => toggleAdmin({ id: user.id, isAdmin: false})}>No</Button>
          </FormControl>
        </HStack>
    )
}