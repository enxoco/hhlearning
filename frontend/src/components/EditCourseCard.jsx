import { DeleteIcon } from '@chakra-ui/icons'
import { Badge, Box, Button, Divider, Flex, FormControl, FormLabel, HStack, IconButton, Input, Stack, Textarea, useColorModeValue, useToast } from '@chakra-ui/react'
import { useState, useRef } from 'react'
import {useRecoilState} from 'recoil'
import {showNewCourseCard} from '../atom'
import { useCreateCourseMutation, useDeleteCourseMutation, useUpdateCourseMutation } from '../generated/graphql'
import Hashids from 'hashids'
const hashids = new Hashids(process.env.REACT_APP_SALT, +process.env.REACT_APP_SALT_LENGTH)

const EditStudentCard = ({name, grade, feedback, id, student, teacher, teacherName}) => {

    const [courseName, setCourseName] = useState(name)
    const [courseGrade, setCourseGrade] = useState(grade)
    const [courseFeedback, setFeedback] = useState(feedback)

    const [updatedCourse, setUpdateCourse] = useUpdateCourseMutation()
    const [createdCourse, createCourse] = useCreateCourseMutation()
    const [_, deleteCourse] = useDeleteCourseMutation()
    const [newCourse, setNewCourse] = useRecoilState(showNewCourseCard)
    const toast = useToast()
    const toastIdRef = useRef()
    const handleCourseNameUpdate = (e) => {
        setCourseName(e.target.value)
    }

    const handleCourseGradeUpdate = (e) => {
        setCourseGrade(e.target.value)
    }

    const handleFeedbackUpdate = (e) => {
      setFeedback(e.target.value)
    }

    const handleFormSubmission = async () => {
        await setUpdateCourse({
            name: courseName,
            grade: courseGrade,
            feedback: courseFeedback,
            teacher: +teacher,
            id: +id
        })
        setNewCourse(false)
        addToast(courseName)
    }

    const handleCreateNewCourse = async () => {
        await createCourse({
            name: courseName,
            grade: courseGrade,
            feedback: courseFeedback,
            student: +hashids.decode(student)[0],
            teacher: +teacher
        })
        setNewCourse(false)
        addToast(courseName)

    }

    const handleDeleteCourse = () => {
      deleteCourse({id})
      addDeletedToast()
    }
    const containerStyle = {
      width: '800px',
      maxWidth: '100%',
      textAlign: 'center'
    }

    function addToast(courseName) {
      toastIdRef.current = toast({
        status: 'success',
        description: `${courseName || 'Course'} saved successfully`,
        position: 'top',
        containerStyle
      })
    }

    function addDeletedToast(courseName) {
      toastIdRef.current = toast({
        status: 'error',
        description: `${courseName || 'Course'} deleted successfully`,
        position: 'top',
        containerStyle
      })
    }
    return (
        <Box as="form" bg="bg-surface" boxShadow={useColorModeValue("sm", "sm-dark")} borderRadius="lg" data-label="course-card">
          <HStack justifyContent="space-between">
            <Badge borderRadius={0} p={5}>Teacher: {teacherName}</Badge>
          </HStack>

        <Stack spacing="5" px={{ base: "4", md: "6" }} py={{ base: "5", md: "6" }}>
          
          <Stack spacing="6" direction={{ base: "column", md: "row" }}>
            <FormControl id="courseName">
              <FormLabel>Course Name</FormLabel>
              <Input defaultValue={courseName} onChange={handleCourseNameUpdate} />
            </FormControl>
            <FormControl id="courseGrade" onChange={handleCourseGradeUpdate}>
              <FormLabel>Grade</FormLabel>
              <Input defaultValue={courseGrade} />
            </FormControl>

          </Stack>
          <FormControl id="feedback">
              <FormLabel>Feedback</FormLabel>
              <Textarea onChange={handleFeedbackUpdate} defaultValue={courseFeedback || null} ></Textarea>
            </FormControl>
        </Stack>
        <Divider />
        <Flex direction="row-reverse" py="4" px={{ base: "4", md: "6" }}>
          <Button variant="primary" onClick={(id) ? handleFormSubmission : handleCreateNewCourse} isLoading={updatedCourse.fetching || createdCourse.fetching} data-action="save-course">Save</Button>
          <IconButton icon={<DeleteIcon />} variant="ghost" mr={'auto'} aria-label='Delete Course' colorScheme="red" onClick={handleDeleteCourse} data-action="delete-course">Delete</IconButton>

        </Flex>
      </Box>
    )
}
export default EditStudentCard