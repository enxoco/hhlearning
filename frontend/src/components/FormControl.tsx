
import { Button, FormControl, FormLabel, HStack, Input } from "@chakra-ui/react"

import { useState } from 'react'
import { useUpdateSettingsMutation } from "#/generated/graphql"

const FormBlock = ({ id, name, value }) => {
    const [setting, updateSetting] = useUpdateSettingsMutation()
    const [initial, setInitial] = useState(value)

    const handleSaveForm = (e) => {
        e.preventDefault()
        updateSetting({ id, value: initial })
    }
    return (
        <FormControl>
            <FormLabel>{name}</FormLabel>
            <HStack>
                <Input name={name} value={initial} onChange={e => setInitial(e.target.value)} />
                <Button onClick={handleSaveForm}>Save</Button>
            </HStack>
        </FormControl>
    )
}

export default FormBlock