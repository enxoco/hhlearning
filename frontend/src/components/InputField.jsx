import {FieldConfig, FieldHookConfig, useField} from 'formik'
import { FormControl, FormLabel, FormErrorMessage } from '@chakra-ui/form-control'
import { Input } from '@chakra-ui/input'
import { Textarea } from '@chakra-ui/textarea'
import { ComponentWithAs, InputProps, TextareaProps } from '@chakra-ui/react'
import { ClassAttributes, InputHTMLAttributes } from 'react'
type InputFieldProps = {
    label: string;
    textarea: boolean;
    props: string | FieldConfig<any>;
}
export const InputField = ({label, textarea, ...props}: InputFieldProps) => {
    let InputOrTextArea: ComponentWithAs<"textarea", TextareaProps> | ComponentWithAs<"input", InputProps> = Input
    if (textarea) {
        InputOrTextArea = Textarea
    }
    //eslint-disable-next-line
    const [field, { error, }] = useField(props)
    return (
        <FormControl isInvalid={!!error}>
        <FormLabel htmlFor={field.name}>{label}</FormLabel>
        <InputOrTextArea {...field} {...props} id={field.name} placeholder={props.placeholder} />
        <FormErrorMessage>{error}</FormErrorMessage>
      </FormControl>
    );
}
