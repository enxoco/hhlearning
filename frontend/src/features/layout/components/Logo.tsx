import { chakra, HTMLChakraProps } from '@chakra-ui/react'
import logo from "#/logo.png"
export default function (props: HTMLChakraProps<'img'>) {
  return (
    <chakra.img
      src={logo}
      alt="Hilger Higher Learning Logo"
      width={300}
      {...props}
    />
  )
}

