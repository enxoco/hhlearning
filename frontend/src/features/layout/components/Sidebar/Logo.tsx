import { chakra, HTMLChakraProps } from '@chakra-ui/react'
import logo from "#/logo.png"
export const Logo = (props: HTMLChakraProps<'img'>) => {
  return (
    <chakra.img
      src={logo}
      alt="Hilger Higher Learning Logo"
      width={300}
      {...props}
    />
  )
}

