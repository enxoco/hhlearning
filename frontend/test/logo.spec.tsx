import React from "react";
import renderer from "react-test-renderer";
import { Logo } from "../src/components/Logo";

function toJson(component: renderer.ReactTestRenderer) {
  const result = component.toJSON();
  expect(result).toBeDefined();
  expect(result).not.toBeInstanceOf(Array);
  return result as renderer.ReactTestRendererJSON;
}

test("Link changes the class when hovered", () => {
  const component = renderer.create(
    <Logo />
  );
  let tree = toJson(component);
  expect(tree).toMatchSnapshot();

});
