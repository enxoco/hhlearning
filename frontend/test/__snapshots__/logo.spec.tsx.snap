// Vitest Snapshot v1

exports[`Link changes the class when hovered 1`] = `
<img
  alt="Hilger Higher Learning Logo"
  className="css-1v5z18m"
  src="/src/logo.png"
/>
`;
